service/history/workflow/update/update_test.go

```diff
@@ -336,6 +337,82 @@ func TestRequestSendReject(t *testing.T) {
 	})
 }
 
+func TestAdmitAcceptOrReject(t *testing.T) {
+	// See comment in onAcceptanceMsg about stateAdmitted.
+	// This test is to ensure that transition from stateAdmitted to stateAccepted or stateCompleted (rejected) is valid.
+
+	t.Parallel()
+	var (
+		ctx     = context.Background()
+		effects = effect.Immediate(ctx)
+		meta    = updatepb.Meta{UpdateId: t.Name() + "-update-id"}
+		req     = updatepb.Request{Meta: &meta, Input: &updatepb.Input{Name: t.Name()}}
+		acpt    = protocolpb.Message{Body: mustMarshalAny(t, &updatepb.Acceptance{
+			AcceptedRequestMessageId:         "random",
+			AcceptedRequestSequencingEventId: 2208,
+		})}
```

Comment:
nit: the `rej` is inside the test case, so arguably `acpt` should probably be, too